platforms/core-configuration/configuration-cache/src/integTest/groovy/org/gradle/internal/cc/impl/ConfigurationCacheGracefulDegradationIntegrationTest.groovy

```diff
@@ -44,6 +45,86 @@ class ConfigurationCacheGracefulDegradationIntegrationTest extends AbstractConfi
 
     def configurationCache = newConfigurationCacheFixture()
 
+    def "should not fail evaluating lazy properties in Groovy in graceful degradation mode"() {
+        def configurationCache = newConfigurationCacheFixture()
+
+        given:
+        // any way to trigger graceful degradation will do
+        settingsFile """
+            sourceControl {
+                vcsMappings {
+                    withModule("org.test:sourceModule") {
+                        from(GitVersionControlSpec) {
+                            url = "some-repo"
+                        }
+                    }
+                }
+            }
+        """
```

Comment:
💅 maybe extract this into a method? Then you can reuse it for the next test too (you can have `settings.gradle` and `build.gradle.kts` together)